lib/view/theme_picker.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:adaptive_theme/adaptive_theme.dart';
+import 'package:flutter/material.dart';
+
+/// ThemePicker:
+///   - a widget which allows the user to change the theme (light/dark mode)
+class ThemePicker extends StatefulWidget {
+  const ThemePicker({super.key});
+
+  @override
+  State<ThemePicker> createState() => _ThemePickerState();
+}
+
+class _ThemePickerState extends State<ThemePicker> {
+  @override
+  Widget build(BuildContext context) {
+    return DropdownMenu(
+        initialSelection: AdaptiveTheme.of(context).mode,
+        onSelected: (newMode) {
+          if (newMode != null) {
+            AdaptiveTheme.of(context).setThemeMode(newMode);
+          }
+        },
+        dropdownMenuEntries: const [
```

Comment:
I just checked, the adaptive_theme library does have support for system theme. I would add a "System" theme as the first option in your entries list. It is what most users will want.